gnssanalysis/gn_download.py

```diff
@@ -846,325 +848,183 @@ def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_fol
         list(executor.map(download_file_from_cddis, files, _repeat(ftp_folder), _repeat(output_folder)))
 
 
-# TODO: Deprecate? Only supports legacy filenames
-def download_prod(
-    dates,
-    dest,
-    ac="igs",
-    suff="",
-    f_type="sp3",
-    dwn_src="cddis",
-    ftps=False,
-    f_dict=False,
-    wkly_file=False,
-    repro3=False,
-):
-    """
-    Function used to get the product file/s from download server of choice, default: CDDIS
-
-    Input:
-    dest - destination (str)
-    ac - Analysis Center / product of choice (e.g. igs, igr, cod, jpl, gfz, default = igs)
-    suff - optional suffix added to file name (e.g. _0 or _06 for ultra-rapid products)
-    f_type - file type to download (e.g. clk, cls, erp, sp3, sum, default = sp3)
-    dwn_src - Download Source (e.g. cddis, ga)
-    ftps - Optionally input active ftps connection object
-    wkly_file - optionally grab the weekly file rather than the daily
-    repro3 - option to download the REPRO3 version of the file
-
+def download_product_from_cddis(
+    download_dir: _Path,
+    start_epoch: _datetime,
+    end_epoch: _datetime,
+    file_ext: str,
+    limit: int = None,
+    long_filename: bool = None,
+    analysis_center: str = "IGS",
+    solution_type: str = "ULT",
+    sampling_rate: str = "15M",
+    project_type: str = "OPS",
+    timespan: _datetime.timedelta = _datetime.timedelta(days=2),
+    if_file_present: str = "prompt_user",
+) -> None:
+    """Download the file/s from CDDIS based on start and end epoch, to the download directory (download_dir)
+
+    :param _Path download_dir: Where to download files
+    :param _datetime start_epoch: Start date/time of files to find and download
+    :param _datetime end_epoch: End date/time of files to find and download
+    :param str file_ext: Extension of files to download (e.g. SP3, CLK, ERP, etc)
+    :param int limit: Variable to limit the number of files downloaded, defaults to None
+    :param bool long_filename: Search for IGS long filename, if None use start_epoch to determine, defaults to None
+    :param str analysis_center: Which analysis center's files to download (e.g. COD, GFZ, WHU, etc), defaults to "IGS"
+    :param str solution_type: Which solution type to download (e.g. ULT, RAP, FIN), defaults to "ULT"
+    :param str sampling_rate: Sampling rate of file to download, defaults to "15M"
+    :param str project_type: Project type of file to download (e.g. ), defaults to "OPS"
+    :param _datetime.timedelta timespan: Timespan of the file/s to download, defaults to _datetime.timedelta(days=2)
+    :param str if_file_present: What to do if file already present: "replace", "dont_replace", defaults to "prompt_user"
+    :raises FileNotFoundError: Raise error if the specified file cannot be found on CDDIS
     """
+    # DZ: Download the correct IGS FIN ERP files
```

Comment:
Can remove `DZ: ` as Des's doc is generic enough?

Reply:
Removed `DZ:`